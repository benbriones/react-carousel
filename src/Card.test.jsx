import React from "react";
import { describe, test, expect } from "vitest";
import { render, fireEvent } from "@testing-library/react";
import Card from "./Card";


describe("Card Component", function () {
    test("renders without crashing", function () {
        render(<Card
            caption="testCap"
            src="test.png"
            currNum={2}
            totalNum={3} />
        );
    });

    test("matches snapshot", function () {
        const { container } = render(<Card
            caption="testCap"
            src="test.png"
            currNum={2}
            totalNum={3} />
        );

        expect(container).toMatchSnapshot();
    });
});