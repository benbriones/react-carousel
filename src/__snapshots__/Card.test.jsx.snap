// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card Component > matches snapshot 1`] = `
<div>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      testCap
    </h4>
    <img
      alt="testCap"
      class="Card-image"
      src="test.png"
    />
    <small
      class="Card-small"
    >
      Image 
      2
       of 
      3
      .
    </small>
  </div>
</div>
`;
